src/titiler/core/titiler/core/factory.py

```diff
@@ -812,6 +824,299 @@ def wmts(
+            color_formula: Optional[str] = Query(  # noqa
+                None,
+                title="Color Formula",
+                description="rio-color formula (info: https://github.com/mapbox/rio-color)",
+            ),
+            colormap=Depends(self.colormap_dependency),  # noqa
+            render_params=Depends(self.render_dependency),  # noqa
+            reader_params=Depends(self.reader_dependency),
+            env=Depends(self.environment_dependency),
+        ):
+            """
+            Return a WMS query for a single COG.
+
+            GetCapability will generate a WMS XML definition.
+
+            GetMap is mostly copied from titiler.core.factory.TilerFactory.part.part
+            """
+            processed_wms_params = process_wms_params(_request)
+            print(processed_wms_params)
+
+            # Return a WMS XML
+            if processed_wms_params["request"] == "GetCapabilities":
+
+                base_url = str(_request.url).split("?")[0]
+
+                non_wms_params = [
+                    param for param 
+                    in parse_qsl(str(_request.url))[1:] 
+                    if param[0].lower() not in base_wms_params
+                ]
+
+                wms_url = f"{base_url}?{urlencode(non_wms_params)}"
+
+                # Grab information from each layer provided
+                layers_dict = {}
+                for layer in processed_wms_params["layers"]:
+                    layers_dict[layer] = {}
+                    with rasterio.Env(**env):
+                        with self.reader(layer, **reader_params) as src_dst:
+                            layers_dict[layer]["srs"] = f"EPSG:{src_dst.crs.to_epsg()}"
+                            layers_dict[layer]["bounds"] = src_dst.bounds
+                            layers_dict[layer][
+                                "bounds_wgs84"
+                            ] = src_dst.geographic_bounds
+                            layers_dict[layer]["abstract"] = src_dst.info().json()
+                
+                # Build information for the whole service
+                service_dict = {
+                    "xmin": min([layers_dict[layer]["bounds_wgs84"][0] for layer in layers_dict]),
+                    "ymin": min([layers_dict[layer]["bounds_wgs84"][1] for layer in layers_dict]),
+                    "xmax": max([layers_dict[layer]["bounds_wgs84"][2] for layer in layers_dict]),
+                    "ymax": max([layers_dict[layer]["bounds_wgs84"][3] for layer in layers_dict]),
+                }
+
+                template = f"wms_{processed_wms_params.get('version')}.xml"
+
+                return templates.TemplateResponse(
+                    template,
+                    {
+                        "request": _request,
+                        "request_url": wms_url,
+                        "formats": [v.value for v in list(WMSMediaType)],
+                        "available_epsgs": [e.value for e in list(EPSGCodes)],
```

Comment:
in theory all EPSG code found in proj are available not just the ones in morecantile 🤷